vj4/handler/user.py

```diff
@@ -253,3 +254,35 @@ def modify_udoc(self, udict, key):
     for i in range(len(udocs)):
       self.modify_udoc(udocs, i)
     self.json(udocs)
+
+
+async def render_or_json_rank(self, page, ppcount, pcount, pdocs, **kwargs):
+  if 'page_title' not in kwargs:
+    kwargs['page_title'] = self.translate(self.TITLE)
+  if 'path_components' not in kwargs:
+    kwargs['path_components'] = self.build_path((self.translate(self.NAME), None))
+  if self.prefer_json:
+    list_html = self.render_html('partials/rank.html', page=page, ppcount=ppcount,
+                                 pcount=pcount, pdocs=pdocs, psdict=psdict)
+    path_html = self.render_html('partials/path.html', path_components=kwargs['path_components'])
+    self.json({'title': self.render_title(kwargs['page_title']),
```

Comment:
Please do not copy the code from problem list's handler and please refer to other lists.

The problem list's handler has something special.